### PR TITLE
setup canvas on first capture screenshot.js

### PR DESCRIPTION
**Pull request by arpu**
**Description:**
The  default Screenshot Component always creates a shader and canvas on init()
this PR change it on first usage of screenshot capture.

**Changes proposed:**
-set canvas on first capture usage

### CODE REVIEW

File: src/components/scene/screenshot.js [hunk 1]
```diff
@@ -51,36 +51,32 @@ module.exports.Component = registerComponent('screenshot', {
   },
 
   init: function () {
+    this.setupDone = false;
```
Comment (dmarcos) [inline src/components/scene/screenshot.js:54]: don't need an extra flag. Can just check if there's a `self.canvas` in the setup function

File: src/components/scene/screenshot.js [hunk 2]
```diff
@@ -178,6 +174,9 @@ module.exports.Component = registerComponent('screenshot', {
    * Maintained for backwards compatibility.
    */
   capture: function (projection) {
+    if (!this.setupDone) {
```
Comment (dmarcos) [inline src/components/scene/screenshot.js:177]: can just do `this.setup()` and check for `this.canvas` inside setup

File: src/components/scene/screenshot.js [hunk 3]
```diff
@@ -51,16 +51,12 @@ module.exports.Component = registerComponent('screenshot', {
   },
 
   init: function () {
```
Comment (dmarcos) [inline src/components/scene/screenshot.js:53]: We can probably remove the init function now

File: src/components/scene/screenshot.js [hunk 4]
```diff
@@ -178,6 +174,7 @@ module.exports.Component = registerComponent('screenshot', {
    * Maintained for backwards compatibility.
    */
   capture: function (projection) {
+    this.setup();
```
Comment (dmarcos) [inline src/components/scene/screenshot.js:177]: Can add this call after variables declaration

**Comment by dmarcos** [comment]
Thank you!

**Comment by mrxz** [comment]
@arpu This PR causes a regression when people use the `screenshot` component in the following way:

```JS
document.querySelector('a-scene').components.screenshot.getCanvas('perspective');
```
The `getCanvas` is part of its public API and documented in case the user doesn't want to automatically download the file, but do something else with it. See https://aframe.io/docs/1.5.0/components/screenshot.html#methods

**Pull request merged**